meta-cli/src/cli/serialize.rs

```diff
@@ -100,39 +107,39 @@ impl Action for Serialize {
             }
         }
 
-        if let Some(prefix) = self.prefix.as_ref() {
-            for tg in loaded.iter_mut() {
-                tg.meta.prefix = Some(prefix.clone());
-            }
-        }
-
-        let tgs = loaded;
-
-        if let Some(tg_name) = self.typegraph.as_ref() {
-            if let Some(tg) = tgs.iter().find(|tg| &tg.name().unwrap() == tg_name) {
-                self.write(&self.to_string(&tg)?).await?;
-            } else {
-                let suggestions = tgs
-                    .iter()
-                    .map(|tg| tg.name().unwrap())
-                    .collect::<Vec<_>>()
-                    .join(", ");
-                bail!(
-                    "typegraph \"{}\" not found; available typegraphs are: {suggestions}",
-                    tg_name
-                );
-            }
-        } else if self.unique {
-            if tgs.len() == 1 {
-                self.write(&self.to_string(&tgs[0])?).await?;
-            } else {
-                bail!("expected only one typegraph, got {}", tgs.len());
-            }
-        } else {
-            self.write(&self.to_string(&tgs)?).await?;
-        }
-
-        Ok(())
+        // if let Some(prefix) = self.prefix.as_ref() {
+        //     for tg in loaded.iter_mut() {
+        //         tg.meta.prefix = Some(prefix.clone());
+        //     }
+        // }
+
+        // let tgs = loaded;
+
+        // if let Some(tg_name) = self.typegraph.as_ref() {
+        //     if let Some(tg) = tgs.iter().find(|tg| &tg.name().unwrap() == tg_name) {
+        //         self.write(&self.to_string(&tg)?).await?;
+        //     } else {
+        //         let suggestions = tgs
+        //             .iter()
+        //             .map(|tg| tg.name().unwrap())
+        //             .collect::<Vec<_>>()
+        //             .join(", ");
+        //         bail!(
+        //             "typegraph \"{}\" not found; available typegraphs are: {suggestions}",
+        //             tg_name
+        //         );
+        //     }
+        // } else if self.unique {
+        //     if tgs.len() == 1 {
+        //         self.write(&self.to_string(&tgs[0])?).await?;
+        //     } else {
+        //         bail!("expected only one typegraph, got {}", tgs.len());
+        //     }
+        // } else {
+        //     self.write(&self.to_string(&tgs)?).await?;
+        // }
+
+        exit(0); // kill the server
```

Comment:
You could gracefully shutdown the server, if needed:
`HttpServer::run` returns a [`Server`](https://docs.rs/actix-web/latest/actix_web/dev/struct.Server.html) instance from which you can get a [`ServerHandle`](https://docs.rs/actix-web/latest/actix_web/dev/struct.ServerHandle.html).

Reply:
Sorry for the late reply, but this can get a bit tricky as the handle needs to be available globally.
Hopefully an implementation like `ServerStore` should make it work.

Reply:
Update: making the Server available prior running `cmd.run` made it possible 👍